### DISCUSSION TITLE
2FA for Login

### DISCUSSION THREAD

**Discussion by sharjeelz** [laravel/nova-issues#4708] [Q&A] [answered]
Hi , Now a days we see many application provide you 2FA, I would like to see this in nova.
Suppose I am creating a user account for someone else, and I want to make sure it remains safe, so a 2FA (like sending OTP through s SMS provider would be great)

please correct me if there is a plugin for this already

**Comment by juse-less** [comment, marked as answer]
You can use custom login pages (and backends) by configuring them in the Nova config.
You have options for middlewares, and a guard, where you could use something like Fortify.
In my case, since I bootstrapped my app with Jetstream, I can directly use their pages by adding the following to my Nova config
```php
'routes' => [
    'login' => '/login',
    'logout' => '/logout',
],
```

It'll then use Jetstream's login pages and logic, where I have then activated 2FA, among others.
Laravel's first-party 2FA packages don't support SMS at this time AFAIK, though.
But maybe it could get you in the right direction.
